daliuge-translator/setup.py

```diff
@@ -104,6 +104,7 @@ def package_files(directory):
     "pyswarm",
     "python-multipart",
     # "ruamel.yaml.clib<=0.2.2",
+    "rfc3986",
```

Comment:
**question**: Any reason you pulled in a seperate library rather than using `urllib.urlparse` which is bundled with python?

**suggestion**: Replace with [urllib](https://docs.python.org/3/library/urllib.parse.html)

Reply:
I think this was a problem with my development environment - 'twas claiming urllib didn't support parse, which is in fact false :laughing:

Reply:
So, looking at this once more, adding this dependency and *only* adding this dependency causes the tests to pass.
I have a sneaky suspicion that this relates to the IPV6 issues Andreas had elsewhere, and me changing the test URL from 127.0.0.1 to localhost.

Reply:
Possibly something that could be happening is `localhost` is resolving to `::1` and we pass the IP directly in somewhere but we're not surrounding the IPv6 address in square brackets as is required by the [RFC](https://datatracker.ietf.org/doc/html/rfc2732.html#section-2).

Reply:
I think I've got this one worked out. :smile;